### PR TITLE
Use QCommandLineParser instead of raw argv inspection

### PR DESCRIPTION
**Pull request by emlun**
This also adds `-v|--version` and `-h|--help` options.

### CODE REVIEW

File: ykman-gui/main.cpp [hunk 1]
```diff
@@ -60,11 +61,22 @@ int main(int argc, char *argv[])
 
     engine.load(QUrl(url_prefix + main_qml));
 
-    if (argc > 2 && strcmp(argv[1], "--log-level") == 0) {
-        if (argc > 4 && strcmp(argv[3], "--log-file") == 0) {
-            QMetaObject::invokeMethod(engine.rootObjects().first(), "enableLoggingToFile", Q_ARG(QVariant, argv[2]), Q_ARG(QVariant, argv[4]));
+    QCommandLineParser parser;
+    parser.setApplicationDescription("Cross-platform application for YubiKey configuration");
```
Comment (dagheyman) [inline ykman-gui/main.cpp]: Add a . in the end.

File: ykman-gui/main.cpp [hunk 2]
```diff
@@ -60,11 +61,22 @@ int main(int argc, char *argv[])
...
+    parser.addHelpOption();
+    parser.addVersionOption();
+    parser.addOptions({
+        {"log-level", QCoreApplication::translate("main", "Set log level to <LEVEL>"), QCoreApplication::translate("main", "LEVEL")},
```
Comment (dagheyman) [inline ykman-gui/main.cpp]: Maybe print out the different log-levels, similar to what `ykman -h` does?

Comment (emlun) [reply ykman-gui/main.cpp]: Good idea! I'll see if Qt has an API for that, or if we'll have to do it by hand.

File: ykman-gui/main.cpp (hunk 1 above)
Comment (emlun) [reply ykman-gui/main.cpp]: Hm. This is just a phrase, not a full sentence. But comparing with `ykman` and a few Unix toolbox programs, it seems like an imperative sentence is the norm. How about rewriting this to "Configure your YubiKey using a graphical application."?

Comment (dagheyman) [reply ykman-gui/main.cpp]: Sounds good!

**Comment by emlun** [comment]
I think the `Cannot convert` warning is unrelated, I've seen it before. I don't know what causes it, though - I haven't seen it in yubioath-desktop...

File: ykman-gui/main.cpp (hunk 2 above)
Comment (emlun) [reply ykman-gui/main.cpp]: There seems to be no API for allowed values, so I just hard-coded them into the option description. They're unlikely to change anyway.

File: ykman-gui/main.cpp (hunk 1 above)
Comment (emlun) [reply ykman-gui/main.cpp]: Done!

**Pull request merged**